test/conformance/tests/sriov_operator.go

```diff
@@ -1093,43 +1093,86 @@ var _ = Describe("[sriov] operator", func() {
 
 			Context("MTU", func() {
 				BeforeEach(func() {
-					node := sriovInfos.Nodes[0]
-					intf, err := sriovInfos.FindOneSriovDevice(node)
-					Expect(err).ToNot(HaveOccurred())
+					var node string
+					resourceName := "mturesource"
+					numVfs := 5
+					var intf *sriovv1.InterfaceExt
+					var err error
 
-					mtuPolicy := &sriovv1.SriovNetworkNodePolicy{
-						ObjectMeta: metav1.ObjectMeta{
-							GenerateName: "test-mtupolicy",
-							Namespace:    operatorNamespace,
-						},
+					if discovery.Enabled() {
+						node, resourceName, numVfs, intf, err = discovery.DiscoveredResources(clients,
+							sriovInfos, operatorNamespace,
+							func(policy sriovv1.SriovNetworkNodePolicy) bool {
+								if !defaultFilterPolicy(policy) {
+									return false
+								}
+								if policy.Spec.Mtu != 9000 {
+									return false
+								}
+								return true
+							},
+							func(node string, sriovDeviceList []*sriovv1.InterfaceExt) (*sriovv1.InterfaceExt, bool) {
```

Comment:
Can this just be "filterDevice" as a function that accepts a single device and returns true if we want it?
I think it would be much more readable.

Reply:
I would prefer not to.
This would require a change for other cases we use it, for example to find the unused devices. That is done by creating a pod and checking all the devices on the single pod. If we modify it, we would have to test each device one by one, with a new pod for each device. This might make a real difference in the test times.

Reply:
Agree. Leave as it is